engine/src/main/java/org/hibernate/search/query/engine/impl/HSQueryImpl.java

```diff
@@ -238,7 +238,7 @@ private TimeoutManagerImpl getTimeoutManagerImpl() {
 			if ( luceneQuery == null ) {
 				throw new AssertionFailure( "Requesting TimeoutManager before setting luceneQuery()" );
 			}
-			timeoutManager = new TimeoutManagerImpl( luceneQuery, timeoutExceptionFactory, this.extendedIntegrator.getTimingSource() );
+			timeoutManager = new TimeoutManagerImpl( String.valueOf( luceneQuery ), timeoutExceptionFactory, this.extendedIntegrator.getTimingSource() );
```

Comment:
I think we should avoid invoking the toString() this early, as the code in many Query implementations is a bit stressing.

Could we make it lazy? Worst case just accepting the 'Object' type would do, and we invoke toString() on that as necessary.. but maybe you have a more elegant idea which doesn't take you more than 5 minutes?